imap/http_caldav_sched.c

```diff
@@ -189,14 +189,15 @@ struct address_t {
 static void add_address(struct address_t **recipients, icalproperty *prop,
                         const char* (*icalproperty_get_address)(icalproperty *))
 {
-    struct address_t *new = xzmalloc(sizeof(struct address_t));
+    struct address_t *new;
```

Comment:
couldn't we move this declaration to its initialization in line 200?

Reply:
Yes, but I don't want to, because:

> the goto done jump in line 3023 could jump over the ptrarray_new in line

This sort of thing is a pain to keep track of, and we've fixed enough goto-skips-declaration and goto-skips-initialisation bugs in recent years that I very strongly dislike having declarations littered throughout functions in the C++ style.  They're cute to write in the first place but they're a nuisance to maintain over the long term.

Are there gotos used in this particular case?  No.  But a good habit of declaring variables at the top of the nearest appropriate scope, and then cleaning up (if necessary) at the end of that scope, always avoids this problem, and also makes verifying proper memory management easier.

It's also the usual C way of doing things -- this isn't perl, javascript, or C++, and just because C99 copied a feature from those languages, doesn't make it best practice to use that feature.  (It's also a terrible habit in C++: it only makes sense in languages with garbage collection or other automatic memory management, which neither C nor C++ are.)

Reply:
Fine with me. I'm not entirely sold on the idea that one or the other way is the "usual C way" (is there any, really?), and I see benefit in both. Declaring every variable on top works best for me when there just a few variables in the whole function ([some say](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#functions) if they exceed 5-10 the function is problematic anyway). Unfortunately that doesn't apply to some epic-scale function definitions in the Cyrus code, so at least declaring non-pointer variables at initialization makes sense there. The function at hand is not one of them, though.

Reply:
The proposed changes do correctly resolve the memory leaks.

I prefer having declaration and definition on the same line as this results less code to read, understand and maintain.  The here exposed code deficiencies can be corrected by less code, than currently proposed.  Less code is better. 

The compilers can hint on many places for incorrect usage of `goto`, like reading there not initialized variables.  Adding extra \_\_attiribute__(()) can help the compilers to provide even more hints.  These hints shall be considered more often than currently.

In my opinion the compilers can detect misuse of variables in the goto destinations, thus having extra precautions, which currently solve no problems, is not necessary: initializing `msgrecs` after the second return, and skipping the changes in the goto-destination produces simpler code.

The bottom line is, that the compiler-found deficiencies shall be listened to more often, instead of writing complicated code, which does not solve existing problems.

Considering that my contributions are often ignored, I can also state that my efforts are not respected.  Does claiming for not being respected change anything?  It does not, thus the argument that a unit of efforts is not respected is irrational one.